pymc_extras/statespace/models/DFM.py

```diff
@@ -0,0 +1,811 @@
+
+        if self.exog_flag:
+            if self.shared_exog_states:
+                for i in range(self.k_exog):
+                    shock_names.append(f"exog_shock_{i}.shared")
+            else:
+                for i in range(self.k_exog):
+                    for j in range(self.k_endog):
+                        shock_names.append(f"exog_shock_{i}.endog_{j}")
+
+        return shock_names
+
+    @property
+    def param_dims(self):
+        coord_map = {
+            "x0": (ALL_STATE_DIM,),
+            "P0": (ALL_STATE_DIM, ALL_STATE_AUX_DIM),
+            "factor_loadings": (OBS_STATE_DIM, FACTOR_DIM),
+        }
+        if self.factor_order > 0:
+            coord_map["factor_ar"] = (FACTOR_DIM, AR_PARAM_DIM)
+
+        if self.error_order > 0:
+            coord_map["error_ar"] = (OBS_STATE_DIM, ERROR_AR_PARAM_DIM)
+
+        if self.error_cov_type in ["scalar"]:
+            coord_map["error_sigma"] = ()
+
+        elif self.error_cov_type in ["diagonal"]:
+            coord_map["error_sigma"] = (OBS_STATE_DIM,)
+
+        if self.error_cov_type == "unstructured":
+            coord_map["error_sigma"] = (OBS_STATE_DIM, OBS_STATE_AUX_DIM)
+
+        if self.measurement_error:
+            coord_map["sigma_obs"] = (OBS_STATE_DIM,)
+
+        if self.exog_flag:
+            coord_map["beta"] = (EXOG_STATE_DIM,)
+            if self.exog_innovations:
+                coord_map["beta_sigma"] = (EXOG_STATE_DIM,)
+
+        return coord_map
+
+    @property
+    def data_info(self):
+        if self.exog_flag:
+            return {
+                "exog_data": {
+                    "shape": (None, self.k_exog),
+                    "dims": (TIME_DIM, EXOG_STATE_DIM),
+                },
+            }
+        return {}
+
+    @property
+    def data_names(self):
+        if self.exog_flag:
+            return ["exog_data"]
+        return []
+
+    def make_symbolic_graph(self):
+        if not self.exog_flag:
+            x0 = self.make_and_register_variable("x0", shape=(self.k_states,), dtype=floatX)
+        else:
+            initial_factor_loadings = self.make_and_register_variable(
+                "x0", shape=(self.k_states - self.k_exog_states,), dtype=floatX
+            )
+            initial_betas = self.make_and_register_variable(
+                "beta", shape=(self.k_exog_states,), dtype=floatX
+            )
+            x0 = pt.concatenate([initial_factor_loadings, initial_betas], axis=0)
+
+        self.ssm["initial_state", :] = x0
+
+        # Initial covariance
+        P0 = self.make_and_register_variable(
+            "P0", shape=(self.k_states, self.k_states), dtype=floatX
+        )
+        self.ssm["initial_state_cov", :, :] = P0
+
+        # Design matrix
+        factor_loadings = self.make_and_register_variable(
+            "factor_loadings", shape=(self.k_endog, self.k_factors), dtype=floatX
+        )
+
+        matrix_parts = [factor_loadings]
+
+        # Leaving space for higher-order factors
```

Comment:
Can you make a more detailed comment here about the structure of the matrix? I guess it's ultimately going to have some block structure, like:

```bash
[ A  |  0  ]
[--------]
[ 0  |  B  ]
```
Just explain that A is the factor loadings with shape whatever, and B is whatever with shape whatever. Then you can say in the `factor_order > 1` case "Add the upper-right block of zeros"